### PR TITLE
Add support for fixed height (with minimal cropping)

### PR DESCRIPTION
**Pull request by davidcharlec**
For layouts where you need fixed height rows with still a overall full aspect ratio feel, use the fixedHeight property.
If set to YES, the rowMaximumHeight will be used for all rows, and the images will fit with some cropping applied.

### CODE REVIEW

File: GreedoLayout/GreedoCollectionViewLayout.m [hunk 1]
```diff
@@ -73,50 +74,91 @@ - (void)computeSizesAtIndexPath:(NSIndexPath *)indexPath
...
+    [self.leftOvers addObject:[NSValue valueWithCGSize:photoSize]];
     
-    if (heightThatFits > self.rowMaximumHeight) {
-        // The line is not full, let's ask the next photo and try to fill up the line
-        [self computeSizesAtIndexPath:[NSIndexPath indexPathForItem:(indexPath.item + 1) inSection:indexPath.section]];
+    BOOL enoughContentForTheRow = NO;
+    CGFloat rowHeight = self.rowMaximumHeight;
+    CGFloat widthMultiplier = 1.0;
+    
+    // Calculations for variable height grid
```
Comment (DennyHoang) [inline GreedoLayout/GreedoCollectionViewLayout.m]: Do you mean `Calculations for variable width grid`?

File: GreedoLayout/GreedoCollectionViewLayout.m [hunk 2]
```diff
@@ -73,50 +74,91 @@ - (void)computeSizesAtIndexPath:(NSIndexPath *)indexPath
...
+    if (self.fixedHeight) {
+        CGFloat totalWidth = 0;
+        NSInteger index = 0;
+        for (NSValue *leftOver in self.leftOvers) {
+            CGSize leftOverSize = [leftOver CGSizeValue];
+            CGFloat scaledWidth = ceil((rowHeight * leftOverSize.width) / leftOverSize.height);
+            scaledWidth += interitemSpacing;
+            
+            if ((totalWidth + (scaledWidth * 0.66)) > contentWidth) {
```
Comment (DennyHoang) [inline GreedoLayout/GreedoCollectionViewLayout.m]: Some context of what 0.66 is?

File: GreedoLayout/GreedoCollectionViewLayout.m [hunk 3]
```diff
@@ -73,50 +74,91 @@ - (void)computeSizesAtIndexPath:(NSIndexPath *)indexPath
...
+                // Adding this photo would mean less than 50% of it would be visible
+                enoughContentForTheRow = YES;
+                [self.leftOvers removeObjectAtIndex:index];
+                break;
+            }
+            
+            totalWidth += (scaledWidth);
```
Comment (DennyHoang) [inline GreedoLayout/GreedoCollectionViewLayout.m]: No need for parenthesis

File: GreedoLayout/GreedoCollectionViewLayout.m [hunk 4]
```diff
@@ -73,50 +74,91 @@ - (void)computeSizesAtIndexPath:(NSIndexPath *)indexPath
 
     CGSize  photoSize     = [self.dataSource greedoCollectionViewLayout:self originalImageSizeAtIndexPath:indexPath];
```
Comment (DennyHoang) [inline GreedoLayout/GreedoCollectionViewLayout.m]: That weird spacing, just mentioning it because I noticed

**Comment by anurlybayev** [comment]
You forgot to update the `GreedoLayoutExample`

File: GreedoLayout/GreedoCollectionViewLayout.m (hunk 1 above)
Comment (davidcharlec) [inline GreedoLayout/GreedoCollectionViewLayout.m]: Nope. Variable height :)

File: GreedoLayout/GreedoCollectionViewLayout.m (hunk 2 above)
Comment (davidcharlec) [inline GreedoLayout/GreedoCollectionViewLayout.m]: Fixed the comment

**Pull request merged**